components/raftstore/src/store/worker/read.rs

```diff
@@ -718,21 +714,20 @@ where
                 "delegate_term" => delegate.term,
                 "header_term" => req.get_header().get_term(),
             );
-            self.metrics.rejected_by_term_mismatch += 1;
+            TLS_LOCAL_READ_METRICS.with(|m| m.borrow_mut().reject_reason.term_mismatch.inc());
             return Err(e);
         }
 
         // Check region epoch.
         if util::check_region_epoch(req, &delegate.region, false).is_err() {
-            self.metrics.rejected_by_epoch += 1;
+            TLS_LOCAL_READ_METRICS.with(|m| m.borrow_mut().reject_reason.epoch.inc());
             // Stale epoch, redirect it to raftstore to get the latest region.
             debug!("rejected by epoch not match"; "tag" => &delegate.tag);
             return Ok(None);
         }
 
         let mut inspector = Inspector {
```

Comment:
Seems can be simplified.

Reply:
Can't. Trait `Inspector` needs `DerefMut`, but `CachedReadDelegate` only implements `Deref` as there is an `Arc` in it. To remove the `Arc`, a lot of changes are involved. Won't do it in this PR.